include/envoy/secret/secret_manager.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/api/v2/auth/cert.pb.h"
+#include "envoy/secret/secret.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * A manager for static secrets.
+ *
+ * TODO(jaebong) Support dynamic secrets.
+ */
+class SecretManager {
+public:
+  virtual ~SecretManager() {}
+
+  /**
+   * @param secret a protobuf message of envoy::api::v2::auth::Secret.
+   * @throw an EnovyEception when the secret type is not implemented yet.
+   */
+  virtual void addOrUpdateSecret(const envoy::api::v2::auth::Secret& secret) PURE;
+
+  /**
+   * @param name a name of the TlsCertificateSecret.
+   * @return the TlsCertificate secret. Returns nullptr if the secret is not found.
+   */
+  virtual const TlsCertificateSecretSharedPtr
+  findTlsCertificateSecret(const std::string& name) const PURE;
```

Comment:
Nit: `Secret` seems to be redundant here, perhaps `secretManager->findTlsCertificate()` is meaningful enough?

Reply:
Renamed